src/components/Personalization/createGetPageLocation.js

```diff
@@ -0,0 +1,16 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+export default ({ window }) => () => {
+  const url = window.location.href;
+  return url.substring(url.indexOf(":") + 3);
```

Comment:
@ninaceban This is rather an edge-case, since very few if any customers would be running their prod sites on non-standard ports, but yes, I think the port should be part of surface URI for the web channel.

Reply:
port is part of the surface id, as per [Surfaces pattern matching spec](https://wiki.corp.adobe.com/display/CJM/Surfaces+pattern-matching+implementation) 
Also added a test for this